api/lib/infrastructure/repositories/knowledge-element-repository.js

```diff
@@ -92,12 +77,49 @@ module.exports = {
     return _.groupBy(knowledgeElements, 'competenceId');
   },
 
-  findByCampaignIdAndUserIdForSharedCampaignParticipation({ campaignId, userId }) {
-    return _findByCampaignIdForSharedCampaignParticipation({ campaignId, userId });
+  async findValidatedByCampaignIdAndUserIdForSharedCampaignParticipationInTargetProfile({ campaignId, userId }) {
+    const sharedCampaignParticipations = await knex('campaign-participations')
+      .select('userId', 'sharedAt')
+      .where({ campaignId, isShared: 'true', userId })
+      .limit(1);
+
+    const targetProfileSkillsFromDB = await knex('target-profiles_skills')
+      .select('target-profiles_skills.skillId')
+      .join('target-profiles', 'target-profiles.id', 'target-profiles_skills.targetProfileId')
+      .join('campaigns', 'campaigns.targetProfileId', 'target-profiles.id')
+      .where('campaigns.id', '=', campaignId);
+    const targetProfileSkills = _.map(targetProfileSkillsFromDB, 'skillId');
+
+    if (!sharedCampaignParticipations[0]) {
+      return [];
+    }
+
+    const knowledgeElements = await _findValidatedByUserIdAndLimitDateQuery({ userId: sharedCampaignParticipations[0].userId, limitDate: sharedCampaignParticipations[0].sharedAt });
```

Comment:
Ici on pourrait déstructurer la campaignParticipation pour rendre le code plus lisible (je me base sur le commentaire précédent pour faire cette proposition). Qu'en pensez-vous ?
```suggestion
    const { userId, sharedAt } = sharedCampaignParticipation
    const knowledgeElements = await _findValidatedByUserIdAndLimitDateQuery({ userId, limitDate: sharedAt });
```

Reply:
done

Reply:
done